add-on/chrome/bootstrap.js

```diff
@@ -299,7 +299,9 @@ var WindowListener = {
           }
         });
 
-        this.addMenuItem();
+        if (!PrivateBrowsingUtils.isWindowPrivate(window)) {
+          this.addMenuItem();
+        }
```

Comment:
Good catch, you're totally right we don't need to add those observers (and it stops the tooltip getting changed when it shouldn't be).